# Common String Methods

In [2]:
import pandas as pd


In [3]:
df=pd.read_csv("../datasets/chicago.csv")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32063 entries, 0 to 32062
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Name                    32062 non-null  object
 1   Position Title          32062 non-null  object
 2   Department              32062 non-null  object
 3   Employee Annual Salary  32062 non-null  object
dtypes: object(4)
memory usage: 1002.1+ KB


In [4]:
df["Department"].nunique()

35

In [5]:
df["Department"].count()

32062

For department column there are only 35 unique values , so if we change the type of department to category,
only those 35 unique values will be stored in memory and duplicates will be pointed to that memory location

In [6]:
df["Department"]=df["Department"].astype("category")
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32063 entries, 0 to 32062
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype   
---  ------                  --------------  -----   
 0   Name                    32062 non-null  object  
 1   Position Title          32062 non-null  object  
 2   Department              32062 non-null  category
 3   Employee Annual Salary  32062 non-null  object  
dtypes: category(1), object(3)
memory usage: 784.2+ KB


Now we can see that memory usage is reduced to 784 KB

In [7]:
#Normal string operations in python
"Hello World".upper()

'HELLO WORLD'

In [8]:
"hello world".title()

'Hello World'

In [8]:
len("Hello world")

11

In [9]:
#lets try this with pandas dataframe
df["Department"].upper()

AttributeError: 'Series' object has no attribute 'upper'

In [10]:
df["Department"]=df["Department"].str.title()

In [11]:
df.head()

Name            Position Title        Department  \
0      AARON,  ELVIA J          WATER RATE TAKER       Water Mgmnt   
1    AARON,  JEFFERY M            POLICE OFFICER            Police   
2       AARON,  KARINA            POLICE OFFICER            Police   
3  AARON,  KIMBERLEI R  CHIEF CONTRACT EXPEDITER  General Services   
4  ABAD JR,  VICENTE M         CIVIL ENGINEER IV       Water Mgmnt   

  Employee Annual Salary  
0              $90744.00  
1              $84450.00  
2              $84450.00  
3              $89880.00  
4             $106836.00

In [12]:
df["Name"].str.lower()

0            aaron,  elvia j
1          aaron,  jeffery m
2             aaron,  karina
3        aaron,  kimberlei r
4        abad jr,  vicente m
                ...         
32058     zygowicz,  peter j
32059      zymantas,  mark e
32060    zyrkowski,  carlo e
32061    zyskowski,  dariusz
32062                    NaN
Name: Name, Length: 32063, dtype: object